tests/functional/iac/test_iac_scan_ci.py

```diff
@@ -0,0 +1,82 @@
+import os
+from pathlib import Path
+
+from tests.conftest import _IAC_SINGLE_VULNERABILITY
+from tests.functional.utils import run_ggshield_iac_scan
+from tests.repository import Repository
+
+
+def test_ci_diff_no_vuln(tmp_path: Path) -> None:
+    # GIVEN a repository
+    repository = Repository.create(tmp_path)
+    iac_file = tmp_path / "iac_file.tf"
+    iac_file.write_text(_IAC_SINGLE_VULNERABILITY)
+    repository.add(iac_file)
+    repository.create_commit()
+    # WHEN scanning it
+    result = run_ggshield_iac_scan(
+        *["ci"],
+        cwd=tmp_path,
+        expected_code=0,
+        env=dict(**os.environ, GITLAB_CI="True", CI_COMMIT_BEFORE_SHA="HEAD"),
+    )
+    # THEN a diff scan should be made between current state and ref
+    assert "> No IaC files changed. Skipping." in result.stdout
```

Comment:
The assertion and the comment are not very related IMO.
comments could be improved here.

TBH, I don't understand this test.

Reply:
I updated the comment on those tests.

Reply:
Sorry but I still don't understand. 
1. You create a repo with a commit containing 1 vulnerability
2. You do a CI scan in a non CI environment
3. You have no vulnerabilities

What is the behavior you want to test ? That calling `ci` in a non CI environment doesn't scan anything ?

Reply:
Yes, there is another test below which test than a vulnerability is found. The `ci` is set `GITLAB_CI` env variable (which I believe was not there the moment you post this comment).

Reply:
Thanks! Sounds good to me with the env variables